ui-trackoccupancydiagram/src/components/utils.ts

```diff
@@ -0,0 +1,78 @@
+import { type TickPattern } from './types';
+
+type DrawTextType = {
+  ctx: CanvasRenderingContext2D;
+  text: string;
+  x: number;
+  y: number;
+  color: string;
+  xPosition?: 'left' | 'center' | 'right';
+  yPosition?: 'top' | 'middle' | 'bottom';
+  font?: string;
+  rotateAngle?: number;
+  stroke?: {
+    color: string;
+    width: number;
+  };
+};
+
+export const drawText = ({
+  ctx,
+  text,
+  x,
+  y,
+  color,
+  xPosition = 'left',
+  yPosition = 'bottom',
+  font = '400 12px IBM Plex Sans',
+  rotateAngle,
```

Comment:
you can add a default value of 0 to avoid further veirification

Reply:
done

Reply:
done